src/primitives.coffee

```diff
@@ -85,6 +85,19 @@ class Collection extends ElementType
 
   add: (val) -> @push val
 
+  findElements: (cond) ->
+    elements = @content.reduce (results, el) ->
+      return results.concat([el.findElements(cond)]) if el.elementType() in ['array', 'object']
+      return results.concat([el]) if cond(el)
+      return results
+    , []
+    _.flatten elements, true
+
+  find: (cond) ->
```

Comment:
I changed how this works. Let me know if this is what you had in mind. The first was recursive, but just did not mutate any variables. This new way is recursive, too, but mutates, which is ok here I suppose.

@fosrias I think the function passed into this method should be able to receive the element, the parent element, and maybe depth? Currently, elements know of their children, but cannot know of their parent, which I think should change (additionally, an element could have multiple parents).

@zdne Would you like to see the style of comments used in Drafter? Has that worked well? I haven't wanted to comment much here as this is so early, it could change quite a bit. But I'd like to add something nice.